mindone/models/t5/t5.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+import html
+import logging
+import re
+import urllib.parse as ul
+
+import ftfy
+from bs4 import BeautifulSoup
+
+import mindspore as ms
+from mindspore import Tensor, nn
+
+from mindone.models.t5.t5_modeling import get_t5_encoder, get_t5_tokenizer
+
+logger = logging.getLogger(__name__)
+
+
+class T5Embedder(nn.Cell):
+    available_models = ["t5-v1_1-xxl"]
+    bad_punct_regex = re.compile(
+        r"["
+        + "#®•©™&@·º½¾¿¡§~"
+        + r"\)"
+        + r"\("
+        + r"\]"
+        + r"\["
+        + r"\}"
+        + r"\{"
+        + r"\|"
+        + r"\\"
+        + r"\/"
+        + r"\*"
+        + r"]{1,}"
+    )  # noqa
+
+    def __init__(
+        self,
+        cache_dir,
+        use_text_preprocessing=True,
+        model_max_length=120,
+        pretrained_ckpt=None,
+    ):
+        super().__init__()
+        self.use_text_preprocessing = use_text_preprocessing
+        self.cache_dir = cache_dir
+        self.pretrained_ckpt = pretrained_ckpt
+
+        self.tokenizer = get_t5_tokenizer(cache_dir)
```

Comment:
I think tokenizers and embedders should be separated into different modules since they represent different steps in the pipeline. Furthermore, tokenizers operate on strings rather than tensors, so it's preferable to keep them outside of the `nn.Cell`.